### ISSUE TITLE
ANNOUNCEMENT: Fen Light Version 1.0.09

### ISSUE THREAD

**Issue by Tikipeter** [Tikipeter/repository.tikipeter#362]
A zip of Fen Light version `1.0.09` has been added to the tikipeter io repo.

This will show up within Kodi the same as the repo zip currently does when installing it. Therefore, you can install/update Fen Light by selecting this zip file.

From then on, Fen Light will be able to update itself, independent of Kodi's repository and addon management system.

The zip for Fen Light will reside away from github, for various (non threatening) reasons.

Moving forward, Fen Light will continue to not be housed within the main Tikipeter repository, as mentioned yesterday. This is due to excessive requests for the bringing back of features/settings, including arguing aggressively in private messages when I declined requests. I can handle people like that, I've been doing that for over 6 years, but I do detest people like that and I don't want to provide them with something I have made anymore if I can help it.

There are, however, many fine users, some of whom may want to use Fen Light. So maybe this can be a compromise.

As Fen Light was forked and made for myself, I will be basing all decisions on it's development on what I personally wish to see in the addon. In leaving Fen available, I think that's fair and I thought it was fair when originally done.

Moving forward, I will be leaving Github entirely, so a solution for Fen will also need to be found.

I will leave comments open for any questions needing to be answered.

NO "THANKYOU" ETC POSTS, JUST ANY QUESTIONS.

**Comment by rowel07** [comment]
The new update system is working well. Thank you for still sharing your work to us. 😘

**Comment by ivarbrandt** [comment]
This is probably off topic but I've been meaning to ask you about this relatively new statement in your readme:

**Installations of Fen/Fen Light on the Android OS are not supported.**

What do you mean by this? I have a Shield Pro that runs Android so I need to know if you're planning on nuking Android functionality 😰

**Comment by Polgara01** [comment]
Question about the new update system: How will we receive updates?
- Only when we check manually
- Automatically
- Fen Light will notify when an update is available

Fen Light results question: I am getting slightly different results when searching for one particular obscure French show.  Fen Light is finding the same EN results but is not finding the RD and PM results that Fen finds.  Is this an expected result or something that should be reported to you?  I'm done watching the show, so this is a zero impact issue for me, just something I thought you might be interested in if it is unexpected.

I know you don't want thanks, so I'll refrain.  I did want to say I enjoyed the limited interactions I've had with you.  Too bad there is no effective way to filter legitimate bug reports from feature requests and other unwanted feedback.  As I mentioned to you before, my gaming addon is my personal addon which I share because it is zero impact on me.  I have comments turned off and delete anything that isn't a bug report or an idea I like.  I hope you can find a way to do something similar so those of use who have legitimate reasons to contact you don't lose all access, but after 6 years I understand if you need a vacation from users.

Take care :)

**Comment by Tikipeter** [comment]
>The new update system is working well.

Good to hear.

>Installations of Fen/Fen Light on the Android OS are not supported.

Just means no support for Android devices. Fen/Light will still work on Android.

>Question about the new update system: How will we receive updates?

Once you have installed version 1.0.09 from the zip file I dropped in the io repo, you can check for and receive updates through Tools->Check for Updates within Fen Light. If an update is available, you can select OK to update at that point. As a test, I have an update already available once you do this, to version 1.0.11.

In the near future, once I'm sure the update code is working correctly, I will include a service that will check for a new update on each Kodi startup. But I need to see the code is working OK first before making it more automatic.

The good from this is that there is no need anymore to install a repository and have the addon update through this. The addon will be able to update itself. The bad is, after I leave github, there will be no io address to put into Kodi's file manager as an easy way to install an addon from zip. So a new address to store the zip will need to be found and shared.

**Comment by AT5Pin** [comment]
Worked just fine for me as your stuff always does. took me 1/2 hour to find my git log in though :) a4k has banned me so many times for sticking up for those who deserved it, that I no longer waste my time in there. Besides, I think they were killing my good karma since 2016 LOL. I know a lot of simple older folks are grateful right now. Those peoples I love helping! Take cares man. I'm out.

**Comment by Tikipeter** [comment]
Thanks for taking the time to report back. Much appreciated.

**Comment by jorgitoxp** [comment]
Tikipeter,

thank you so much for giving us (the ones that really appreciate your work) the privilege to keep using fen light.  I knew you would somehow find a way to share light, because that is the type of person you are.  At least that is what I’ve gather from months on following you on Reddit.  I had you as a favorite and multiple times a day would read your comments, because I would learn tremendous amounts of information from your insight helping other users.  Thank you again for sharing your hobby with me, wherever you take feb light I will ride with it, because this is your personal add on and we are all taking advantage of using it for free.  

Take care of your self, and just block the outside noise.  Like you said, there are many good and thankful users out there, those are the ones that matter.  

Blessings and happy holidays!

**Comment by les017** [comment]
Can confirm perfect and, as expected, super clean and fast update to .11.  Everything is so much appreciated!

**Comment by Tikipeter** [comment]
>Can confirm

Thankyou les917. Much appreciated reporting back.

**Comment by jorgitoxp** [comment]
And also can confirm fen light internal updater worked like a charm.  It updated to version .11 super fast.  Thanks you again!

**Comment by Tikipeter** [comment]
>And also can confirm fen light internal updater worked like a charm.

Thanks a lot jorgitoxp for taking the time to confirm.

**Comment by derrgoo** [comment]
All good and updated smooth! Thanks 🙏

**Comment by ivarbrandt** [comment]
You know I don't use FEN Light but I had to try the built in updater. 

Oowee that built in updater is fancy man. Updated without a hitch. Good stuff.

**Comment by Tikipeter** [comment]
All good and updated smooth! 
Thanks derrgoo for taking the time. You're appreciated.

>You know I don't use FEN Light but I had to try the built in updater.
Oowee that built in updater is fancy man. Updated without a hitch. Good stuff.

Thanks buddy. Maybe one day you'll come over. :)

**Comment by Cloudman999** [comment]
Hi @Tikipeter 
I was able to update Fen Light from within to .11 using the inbuilt updater.

Quick question - if I keep a copy of the last .zip version of Fen Light plugin, can I just reinstall it within Kodi (install from .zip), and then use internal updater to get future updates?

Regards
@Cloudman999

**Comment by Tikipeter** [comment]
>I was able to update Fen Light from within to .11 using the inbuilt updater.

Thanks for confirming.

>Quick question - if I keep a copy of the last .zip version of Fen Light plugin, can I just reinstall it within Kodi (install from .zip), and then use internal updater to get future updates?

Yes. Any version from version `1.0.09` onwards will be able to update without being associated with a repository.

Hopefully from the next version `1.0.12`, the check for an update will be automatic upon each Kodi start. With the option of going into Tools to check manually as well. Just finishing the code, but I have to test it thoroughly to make sure nothing goes wrong.

**Comment by Cloudman999** [comment]
> > I was able to update Fen Light from within to .11 using the inbuilt updater.
> 
> Thanks for confirming.
> 
> > Quick question - if I keep a copy of the last .zip version of Fen Light plugin, can I just reinstall it within Kodi (install from .zip), and then use internal updater to get future updates?
> 
> Yes. Any version from version `1.0.09` onwards will be able to update without being associated with a repository.
> 
> Hopefully from the next version `1.0.12`, the check for an update will be automatic upon each Kodi start. With the option of going into Tools to check manually as well. Just finishing the code, but I have to test it thoroughly to make sure nothing goes wrong.

Understood @Tikipeter 

Many thanks for your quick reply.

Regards
@Cloudman999

**Comment by xyzfre** [comment]
@Tikipeter update worked perfect and fast Legend ✊🏼

**Comment by Tikipeter** [comment]
>update worked perfect and fast

Great stuff xyzfre. Thanks buddy.

**Comment by sasagr** [comment]
Hi Tiki. For the sake of a clarification I would like to ask you if you intend to maintain Fen in the future at least for future arising compatibility issues.

**Comment by Tikipeter** [comment]
Hey mate.

>maintain Fen in the future at least for future arising compatibility issues.

Yes.

>The fact that fen light will not be hosted in GitHub does it mean that user will not be able to report issues?

Fen won't be hosted on github either. Things aren't properly sorted out yet, but I'll need to get away grom Github pretty soon.

>because I just had an error while trying to authorize traktv. This is fixed now.

Trakt's been a bit slow to respond the last day or so.

**Comment by awsms** [comment]
MEDS, NOW

**Comment by DaleAlan** [comment]
What inbuilt updater witchcraft is this? Integrated works brilliant and everything running smoothly per usual, sorcery I tells ya. With your future move away from GitHub do you have any plans to deploy the new updater system into full fat Fen just in case of eventualities? Absolutely no problem if not as you’ve kept everyone in the loop of wanting to concentrate on Fenlight for months so no biggie. You’ll be a huge miss Peter. Have a cracking Christmas and some more than well deserved you time. All the best for the future.

**Comment by Tikipeter** [comment]
> Integrated works brilliant and everything running smoothly

Appreciate the reply reporting back. Thanks a lot. :)

>...do you have any plans to deploy the new updater system into full fat Fen

Yes, unfortunately the same will have to happen with Fen as there won't be a repository anymore for it.

**Comment by Tikipeter** [comment]
Version `1.0.12` should be available now, with a service to run at startup checking for updates.

Fingers crossed it will work OK for the next update.

**Comment by onlytanmoy** [comment]
@Tikipeter  many thanks mate, currently rocking v1.0.11 and enjoying the blazing fast speed of your amazing add-on. Happy Holidays 😊

**Comment by digitalbarrito** [comment]
> Version `1.0.12` should be available now, with a service to run at startup checking for updates.
> 
> Fingers crossed it will work OK for the next update.

Updated to 1.0.12 without a hitch 👍🏼 Crossing my fingers with you on the next update cause this system is slick 🤞🏼

**Comment by Cloudman999** [comment]
> Version `1.0.12` should be available now, with a service to run at startup checking for updates.
> 
> Fingers crossed it will work OK for the next update.

Hi @Tikipeter 
Update to 1.0.12 worked great.
Cheers
@Cloudman999

**Comment by basicals** [comment]
Auto updated flawlessly on a Shield TV 2017.
Lmao @"Fat Fen" , could be the official name.

**Comment by xyzfre** [comment]
@Tikipeter this new updater is brilliant, I would say can't get better than this, but you're a legend💯 Full of surprises my friend.

**Comment by snvoid** [comment]
This is probably a dumb question as i'm not too very well versed in the kodi addon scene or github, but will fen light / fen still be forkable and be opensource? were the previous ones opensource? 

Thanks

**Comment by Cloudman999** [comment]
Hi @Tikipeter 
Fen light prompted me for an update to version 1.0.13 when Kodi first started. I think it prompted after about 90 seconds. Update appears successful.

Regards
@Cloudman999

**Comment by Tikipeter** [comment]
I've lost track a little of the helpful replies as to the working condition of the updater. So thank you to onlytanmoy, basicals & Cloudman999 for the reports of success. Much appreciated.

>This is probably a dumb question as i'm not too very well versed in the kodi addon scene or github, but does that mean fen light / fen wont be forkable and be opensource? were the previous ones opensource? or is this just my misunderstanding

Both addons will remain open source. A zip of the addon's code is always available from their locations. Which will be gitlab moving forward.

I will need to set up a very basic website to mimic github's io page that can be entered into Kodi's File Manager though. And I've no idea on how to do that, so a little reading will soon be necessary.

**Comment by DaleAlan** [comment]
Loving the new way of updating. On the changelog it’s stated that there’s two settings one of which is the ability to change delay before running time but it’s not showing in the Manage Updates section. Just a heads up though as there’s no issues at all though as it’s worked flawlessly and updated perfectly so I wouldn’t have changed it anywho I was just being nosey :)

**Comment by Cloudman999** [comment]
> I've lost track a little of the helpful replies as to the working condition of the updater. So thank you to onlytanmoy, basicals & Cloudman999 for the reports of success. Much appreciated.
> 
> > This is probably a dumb question as i'm not too very well versed in the kodi addon scene or github, but does that mean fen light / fen wont be forkable and be opensource? were the previous ones opensource? or is this just my misunderstanding
> 
> Both addons will remain open source. A zip of the addon's code is always available from their locations. Which will be gitlab moving forward.
> 
> I will need to set up a very basic website to mimic github's io page that can be entered into Kodi's File Manager though. And I've no idea on how to do that, so a little reading will soon be necessary.

Hi @Tikipeter 
I have a basic idea of how to mimic github's io page. I don't want to provide the details here. Is there a way to send the idea to you? Is there some kind of private DM here on github?

Let me know
@Cloudman999

**Comment by sasagr** [comment]
I updated it to .13 but in settings I see only the “update action” and not the “delay before running”. 
![Uploading IMG_3214.jpeg…]()

**Comment by sasagr** [comment]
![IMG_3214](https://github.com/Tikipeter/repository.tikipeter/assets/3427877/c25d166c-df1c-42a0-8cfa-07571b550ce0)

**Comment by Polgara01** [comment]
Two issues:

1. Same issue as sasagr.  "Update Action" appears. "Delay Before Running" does not appear. Tested on 4 units.

2. Settings->"UTC Time Offset", How do we add a minus sign for UTC -# locations?  I am only seeing a numerical keypad.

Thank you.

**Comment by sasagr** [comment]
> 2. Settings->"UTC Time Offset", How do we add a minus sign for UTC -# locations?  I am only seeing a numerical keypad.
> 

the option to add a - or + will appear after you confirm (Done) the number from the keypad. I had the same question when it happened to me. So maybe it is not idea the way it is.

**Comment by TwilightMercy** [comment]
> I will need to set up a very basic website to mimic github's io page that can be entered into Kodi's File Manager though. And I've no idea on how to do that, so a little reading will soon be necessary.

Doesn't Gitlab offer free Gitlab io page like in github.io?

**Comment by digitalbarrito** [comment]
> I updated it to .13 but in settings I see only the “update action” and not the “delay before running”.

Same thing here.

**Comment by Tikipeter** [comment]
Hey here's a funny one for anyone who wants to check, and doesn't currently have the `Delay Before Running (secs)` setting appearing....... Turn on `Check Volume Before Start of Playback` in the Playback settings and then check if the Delay setting appears. :)

I'll fix it and push an update.

**Comment by Tikipeter** [comment]
>I have a basic idea of how to mimic github's io page. I don't want to provide the details here. Is there a way to send the idea to you? Is there some kind of private DM here on github?

Thanks a lot Cloudman999. I may need to take you up on that offer soon if I can't work something out. Really, the only private options would be Discord or Reddit private message.

>Doesn't Gitlab offer free Gitlab io page like in github.io?

Yes, they definitely do. But I can't make heads or tails of it. With github, you just need an `index.html` file in the repo location, with very basic html. This gitlab one talks about using a YAML file, an HTML file etc. I'll have another read through in a couple of days.

**Comment by DaleAlan** [comment]
> Hey here's a funny one for anyone who wants to check, and doesn't currently have the `Delay Before Running (secs)` setting appearing....... Turn on `Check Volume Before Start of Playback` in the Playback settings and then check if the Delay setting appears. :)
> 
> I'll fix it and push an update.

What the hell. It’s worked haha you should have had a competition to find it.

**Comment by Nunezshow** [comment]
Just went on and updated to 1.0.14, with Update Action and Delay before Running.   Great work

**Comment by Cloudman999** [comment]
Hi @Tikipeter 
Update to version 1.0.14 working in Fen Light.

Cheers
@Cloudman999

**Comment by Cloudman999** [comment]
Hi @Tikipeter 

I created a fork of your existing Github .io page, to try and work out a way to transfer it to a GitLab .io page.

Below is a theoretical way of doing this. I have not tested it - but it may give you some help towards getting it to work.

Here is the location of the fork: https://github.com/Cloudman999/tikipeter.github.io

Here are the basic steps:

1. Export the fork from GitHub as a ZIP file. You can do this by going to the GitHub repository, clicking the "Code" button, and then selecting "Download ZIP". 

2. Create a new project in GitLab and initialize it as a repository (without adding any files yet). 

3. Unzip the exported ZIP file locally. 

4. Add and commit all the files from the local unzipped folder to your new GitLab repository. 

5. Push the commits up to GitLab.

6. In your GitLab project settings, enable Pages by selecting the "Docs" or "Project" option under "Pages". Choose the branch you want to deploy (main most likely). 

7. GitLab will build the site and you should now have your fork deployed at https://yourusername.gitlab.io/project-name

A few things to note:

- The site history will not transfer over, only the latest content.
- You'll need to reconfigure any plugins/themes as needed for the new GitLab Pages environment.
- URLs will change from GitHub.io to GitLab.io 

Again please remember I have not tested this - it is theoretical.

@Cloudman999

**Comment by Cloudman999** [comment]
> > I have a basic idea of how to mimic github's io page. I don't want to provide the details here. Is there a way to send the idea to you? Is there some kind of private DM here on github?
> 
> Thanks a lot Cloudman999. I may need to take you up on that offer soon if I can't work something out. Really, the only private options would be Discord or Reddit private message.
> 
> > Doesn't Gitlab offer free Gitlab io page like in github.io?
> 
> Yes, they definitely do. But I can't make heads or tails of it. With github, you just need an `index.html` file in the repo location, with very basic html. This gitlab one talks about using a YAML file, an HTML file etc. I'll have another read through in a couple of days.

Hi @Tikipeter 

You're right, GitLab Pages does have some additional configuration requirements compared to GitHub Pages. Here are a few key points on setting it up with just an index.html file like GitHub Pages:

- No need for a YAML configuration file unless you want to customize the build process. The default build behavior is similar to GitHub Pages.

- You can simply put an index.html file in the root of your default branch (typically main) and it will get served.

- Make sure to enable Pages for your project/repository in the Settings. 

- Select the branch that contains your index.html file (likely main). 

- GitLab will automatically build the site from that branch and serve it at yourdomain.gitlab.io

- Additional assets/files can also be included directly in the repository like GitHub Pages.

So in summary, for a simple static site with just index.html, GitLab Pages works very similarly to GitHub Pages - just drop the file in and enable Pages. The YAML config is only needed if customizing builds.

@Cloudman999

**Comment by ambulancePilot** [comment]
Hey. Have you clarified how we are to report bugs for Fen?

Secondly, wondering if you had a chance to look at the watched list code I submitted as a fix for the sorting issue and whether it is problematic or good enough to run on my set-up. I don't want any downstream problems from that.

Thanks for you work!

**Comment by onlytanmoy** [comment]
Hi @Tikipeter hope you having a good day so far.
qq - are there any settings or tweaks I can do to make the Easynews files play in one attempt? Most of the times it will go to some retry loop and say I originally clicked on a 4k source, it will retry, then move onto the next, retry and finally plays a 1080p source, so was wondering if you can suggest me some settings in Fen Light or Kodi?

**Comment by ivarbrandt** [comment]
Took a bit but it auto-updated to 1.0.14 without any issues. Thanks man.

**Comment by TwilightMercy** [comment]
> I'm thinking of changing the default behavior for updates to "Notification" so it's a little less obtrusive. In saying that, it would then require the user to go into Tools and update manually. So there's definitely a negative to that.

Out of curiosity, what is the reason for the 90 seconds delay before checking for updates? (And not less like 30 for example)

**Comment by Tikipeter** [comment]
Mainly to cover enough time for users widgets to load, and for the services of addons to finish their initial runs.

If someone with slow hardware and a decent amount of widgets can let me know they'd only need around 30 seconds for all that to happen on startup, then I'd love to change the default for that as well.

There's no SPECIFIC reason that the update code couldn't run as soon as possible, but I can foresee it maybe causing issues so I went the safe route. For now at least.

**Comment by Tikipeter** [comment]
I think you may be right to be honest.

It's an impossible figure to determine though.

You want it to be after initial startup has finished, but before users start actually using the device. I could easily add a check to make sure at least Fen Light services were finished. Maybe that and a shorter timeout will be the way to go.

**Comment by onlytanmoy** [comment]
> There are, but the best thing you can do to help yourself is to go to the easynews server testing page and find your best route....
> 
> https://members.easynews.com/route/

thanks for your response mate, for some odd reason the test in that page is stuck at loading for me. I tried different browsers, incognito mode, even a different laptop and my android browser! The highest I have seen the loading to go is at 67% and then the never ending spiral wheel with no progress. Same behaviour when I tried the Advanced Test. Maybe its just me :( I even reported this issue to the EN support and they couldn't help with that. They just asked me to try different regions from the available 3 choices depending on my geography. How long the test typically takes at your time?

**Comment by Tikipeter** [comment]
Yeah unfortunately it looks to be getting stuck for me also. I'd wait a day or two and then try it again. They may be having issues with it. Why they'd not tell you that I'm not sure though.

**Comment by onlytanmoy** [comment]
> Yeah unfortunately it looks to be getting stuck for me also. I'd wait a day or two and then try it again. They may be having issues with it. Why they'd not tell you that I'm not sure though.

Thanks for checking at your end, so its just not me alone which indicates some issue at EN end. Yeah, they seemed to be in a rush to close the support ticket somehow by just suggesting some generic troubleshooting like clearing cache/cookies, turning off ad-block, incognito mode etc haha.

**Comment by Cloudman999** [comment]
> @Cloudman999
> 
> I really appreciate the time you spent looking into this for me. I'm hoping to have some free time from work and sleep in about a week's time, after Christmas is over. I will follow what you have written and hopefully that will work. Thanks again.

No trouble at all @Tikipeter 
Like I said, it's just some ideas for you to try out. Hopefully, sone of it will get your GitLab .io page working.

Have a great Xmas break and well deserved rest. Enjoy!

Best regards
@Cloudman999

**Comment by drinfernoo** [comment]
@Tikipeter If the issue is with dealing with users, why not just turn off communication? It's very easy to turn off issues on any particular repo 🤔

**Comment by ivarbrandt** [comment]
> I'm thinking of changing the default behavior for updates to "Notification" so it's a little less obtrusive. In saying that, it would then require the user to go into Tools and update manually.

This is honestly my preference. I always prefer to have control over the updates. A notification lets me know an update is available then I can decide what I want to do. This also works because some people have more/less widgets, addons, faster/slower devices etc.

That being said, any way you choose to go about it is cool with me. I support your decision no matter what you choose.

**Comment by Tikipeter** [comment]
Well, after some effort this morning with enabling and setting up gitlab pages, I have unfortunately had to relent and set up 2FA on this Github account.

Which means I will end up staying on Github. This has been a massive pain in the a** that they have done this. Now Fen Light updates are pointing to gitlab, so I will need to change that over to a github url.

The good thing is it won't affect you users at all. I can simply change the url's needed in the next update and continue on.

**Comment by Tikipeter** [comment]
>This is honestly my preference. I always prefer to have control over the updates. A notification lets me know an update is available then I can decide what I want to do. This also works because some people have more/less widgets, addons, faster/slower devices etc.

Yep, we're thinking the same way.

**Comment by Tikipeter** [comment]
In more bad news, I've had to roll back some of the changes to database access that has been in Fen Light for a while now. These changes helped speed up important parts of the addon, but it also looks like they were causing some pretty serious issues.

Fen Light should still be faster than Fen, but perhaps the difference won't be quite as noticeable, if in fact you are currently noticing a difference.

Sad, but necessary moving forward.

**Comment by derrgoo** [comment]
I have been using light since beginning and the only issue I have noticed is the widgets not loading and just needing a skin reset after that all good. 👍

**Comment by Tikipeter** [comment]
>.....the widgets not loading and just needing a skin reset after that all good.

Yep, that's one of the issues I think to do with the way databases are handled currently.

**Comment by jorgitoxp** [comment]
> I have been using light since beginning and the only issue I have noticed is the widgets not loading and just needing a skin reset after that all good. 👍

Same with me.  I’ve been using light from the beginning, only issue I’ve had is widgets in fentastic not loading sometimes . And not all widgets, some would load but others just shows an icon.

**Comment by xyzfre** [comment]
> In more bad news, I've had to roll back some of the changes to database access that has been in Fen Light for a while now. These changes helped speed up important parts of the addon, but it also looks like they were causing some pretty serious issues.
> 
> 
> 
> Fen Light should still be faster than Fen, but perhaps the difference won't be quite as noticeable, if in fact you are currently noticing a difference.
> 
> 
> 
> Sad, but necessary moving forward.

As a end-user speed has always been perfect even in the past ,Fen to this day, still performs super fast on my end, it's perfect both Fen/Fenlight

**Comment by Cloudman999** [comment]
> Well, after some effort this morning with enabling and setting up gitlab pages, I have unfortunately had to relent and set up 2FA on this Github account.
> 
> Which means I will end up staying on Github. This has been a massive pain in the a** that they have done this. Now Fen Light updates are pointing to gitlab, so I will need to change that over to a github url.
> 
> The good thing is it won't affect you users at all. I can simply change the url's needed in the next update and continue on.

Hi @Tikipeter 
So sorry to hear that GitLab hasn't been successful, and you have had to go back to Github. I guess my suggestions/ideas for setting up your GitLab .io didn't work.

I did some digging around to see if I could find out anything else to help, but it seems GitLab may be trickier to use than GitLab to host your addons. Here are some things to check:

Some specific challenges or limitations that developers may encounter when using GitLab instead of Github...

Here are some specific challenges Kodi add-on developers may face when using GitLab instead of GitHub:

- Package installation - Kodi addons are typically installed directly from GitHub links. GitLab URLs won't work out of the box and a new installation method would need to be implemented.

- Continuous integration - GitHub Actions provide a simple way to set up CI for building/testing addons on each commit. With GitLab, extra configuration is required to set up equivalent pipelines. 

- Peer reviews - GitHub's unified pull requests and review features make code reviews seamless. GitLab PRs require more manual effort like separate issue tracking.

- Versioning - Kodi addons follow semantic versioning practices. GitLab may lack certain release/version management features compared to GitHub. 

- Documentation hosting - GitHub wikis, pages and guides are well integrated. Equivalent long-form documentation hosting on GitLab requires extra configurations.

- Platform support - Some Kodi addon tools like skin compilers assume GitHub and don't support GitLab URLs out of the box. 

- Issue tracking - Standard Kodi issue/bug templates won't work on GitLab by default. New templates need defining.

- Community resources - Forums, tutorials, plugins mostly use GitHub context. Adapting for GitLab could introduce temporary fragmentation.

- Analytics - Useful GitHub metrics on downloads, referrers etc. are not available natively on GitLab during development.

- Feature gaps - Some GitHub features like dependency graphs, archived repos may not have direct equivalents on GitLab yet.

So in short, the integration and ecosystem challenges are the biggest hurdles to using GitLab over GitHub for hosting Kodi addons currently.

Hope this is of use @Tikipeter?

One more thing... Have you checked if the GitLab repository has the necessary info.json file?

Here are some things to check if you're having trouble getting your Kodi add-on installed from a GitLab repository:

- Make sure the GitLab repository contains a valid info.json file with the correct metadata like ID, name, author, version, repo URL.

- Double check the "repository" field in info.json points to the full GitLab URL for the zipball archive, e.g. https://gitlab.com/username/repo/-/archive/master/repo-master.zip 

- Verify the folder structure in GitLab matches what Kodi expects - a zipball of the root folder containing the addon.xml file

- Try installing directly with the zipball URL instead of the whole GitLab page, in case the page embedding causes issues

- Make sure any authentication to a private GitLab repo is working - check with a test OAuth token 

- Check for network/firewall issues blocking access to GitLab domains like .gitlab.com

- Ensure the GitLab project/repo is publicly accessible and not private/internal only

- Compare with a test add-on from GitHub - if that installs, the issue is likely GitLab specific  

- Search the Kodi forums for any known issues with GitLab as an installation source

- Try a different add-on installer method like from ZIP instead of directly from GitLab page

- As a test, change the repo to point to GitHub temporarily to isolate if it's GitLab or add-on code causing issues


@Cloudman999

**Comment by Tikipeter** [comment]
I really appreciated the help Cloudman9999, there were just too many hurdles to overcome. Every instructional write-up I read had the settings/options in places they no longer were, and even then it didn't work when I did thigs like have an empty YAML file. Even though they said it would work, it never did.

I can make a Kodi addon OK, but the stuff they were talking about me having to do were over my head in the end. :)

**Comment by Cloudman999** [comment]
> I really appreciated the help Cloudman9999, there were just too many hurdles to overcome. Every instructional write-up I read had the settings/options in places they no longer were, and even then it didn't work when I did thigs like have an empty YAML file. Even though they said it would work, it never did.
> 
> I can make a Kodi addon OK, but the stuff they were talking about me having to do were over my head in the end. :)

Hi @Tikipeter 
That's OK, no worries. The more I digged into it, the more it became a rabbit hole.

I can understand if it is just too difficult. Maybe it wasn't meant to be.

At least you gave it your best shot, and know more now about the limitations of GitLab.

Maybe that's why most Kodi devs only use Github for their addons. Trying to get the same stuff happening on GitLab isn't where to host your add-on.

Happy to try and do more research if you want, but I'm all out of ideas at the moment, myself.

I know maybe they have forced 2FA on you, but I don't know the ins and outs why it has become an issue for you ( and don't need to know why, as that is your business not mine).

As long as I can still enjoy Fen and Fen Light going forward, and you are still able to improve on Fen Light when/if necessary, I will be eternally grateful.

Much appreciate all you do, mate

Regards
@Cloudman999

**Comment by sasagr** [comment]
I have been using both Fen and Fen light on my two Apple TVs (1080p model and 4K 1st gen). Not noticing a noticeable faster speed in fen light. They are both struggling with the 1080p model and doing pretty well with the 4K loading fentastic skin/widgets in about 5 secs from start. 

Off Topic: @Tikipeter did u ever consider to implement also offcloud as debrid resolver? I got a cheap lifetime sub as investment for my kids 😄

**Comment by Omarscomin** [comment]
> Off Topic: @Tikipeter did u ever consider to implement also offcloud as debrid resolver? I got a cheap lifetime sub as investment for my kids 😄

@sasagr No more features/enhancements are being added to FEN. Only bug fixes and things of that nature. 

FEN Light is...well...a *light* version of FEN, so it seems highly unlikely that he'd want to add that feature to FEN Light. 

Do your research *before* purchasing lifetime subscriptions😄
